controllers/fleetnotification/fleetnotification_controller.go

```diff
@@ -55,21 +63,40 @@ func (r *ManagedFleetNotificationReconciler) Reconcile(ctx context.Context, requ
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling FleetNotification")
 
-	// TODO(user): your logic here
 	nr := ocmagentv1alpha1.ManagedFleetNotificationRecord{}
 
 	err := r.Client.Get(ctx, request.NamespacedName, &nr)
```

Comment:
For a new MC cluster which doesn't have any hosted clusters yet, I think it would be expected to have MFNR as zero. In that case, should the controller still report error? From a point a view, the errors will exist till there's atleast 1 SL notification sent for any hosted cluster. This doesn't seem a big concern but thinking along the lines of when the `Get` request would fail.

Reply:
Yup, agreed. I was just thinking of the time when for a fresh MC cluster we won't have and MFNR created at all during which the controller will fail. But its all expected behaviour. So all good.